homeassistant/components/flexpool/sensor.py

```diff
@@ -0,0 +1,69 @@
+"""Support for Etherscan sensors."""
+from datetime import timedelta
+
+import flexpoolapi
+
+from .sensors import (
+    FlexpoolBalanceSensor,
+    FlexpoolHashrateSensor,
+    FlexpoolPoolHashrateSensor,
+    FlexpoolPoolLuckSensor,
+    FlexpoolPoolWorkersSensor,
+    FlexpoolWorkerHashrateSensor,
+    FlexpoolWorkerShareSensor,
+)
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+
+async def async_setup_entry(hass, entry, async_add_entities, discovery_info=None):
+    """Set up the Etherscan.io sensors."""
+    address = entry.data["address"]
+
+    sensors = [
+        FlexpoolBalanceSensor("flexpool_unpaid_balance", address),
+        FlexpoolHashrateSensor("flexpool_current_reported", address),
+        FlexpoolHashrateSensor("flexpool_current_effective", address),
+        FlexpoolHashrateSensor("flexpool_daily_average", address),
+    ]
+
+    if "workers" in entry.data:
```

Comment:
Please add all available entities. If an entity is less common, disable them by default.